### ISSUE TITLE
[Plugin][Android][Google Sign In] PlatformException: errorCode: sign_in_requiered

### ISSUE THREAD

**Issue by DQvsRA** [flutter/flutter#21178]
## Steps to Reproduce

1. Copy example from https://pub.dartlang.org/packages/google_sign_in#-example-tab-
2. Run on Android

![screen shot 2018-08-29 at 20 56 56](https://user-images.githubusercontent.com/442111/44810081-f5c9f200-abd0-11e8-9c82-2ce22aa33b35.png)

## Logs

```
21:17:02.445 1 info flutter.tools Launching app/main.dart on Android SDK built for x86 in debug mode...
21:17:02.480 2 info flutter.tools Initializing gradle...
21:17:04.514 3 info flutter.tools Resolving dependencies...
21:17:06.277 4 info flutter.tools Running 'gradlew assembleDebug'...
21:17:07.204 5 info flutter.tools registerResGeneratingTask is deprecated, use registerGeneratedResFolders(FileCollection)
21:17:07.205 6 info flutter.tools registerResGeneratingTask is deprecated, use registerGeneratedResFolders(FileCollection)
21:17:07.205 7 info flutter.tools registerResGeneratingTask is deprecated, use registerGeneratedResFolders(FileCollection)
21:17:16.399 8 info flutter.tools Built build/app/outputs/apk/debug/app-debug.apk.
21:17:16.924 9 info flutter.tools Installing build/app/outputs/apk/app.apk...
21:17:22.020 10 info flutter.tools Syncing files to device Android SDK built for x86...
21:17:28.654 11 finer runtime.gc collection time 1ms • 20.3MB used of 21.6MB • isolates/527821612
21:17:29.023 12 finer runtime.gc collection time 2ms • 24.8MB used of 26.1MB • isolates/527821612
21:17:29.262 13 finer runtime.gc collection time 4ms • 28.7MB used of 30.9MB • isolates/527821612
21:17:29.572 14 info flutter.tools D/        (21160): HostConnection::get() New Host Connection established 0x8c82a600, tid 21192
21:17:29.572 15 info flutter.tools D/EGL_emulation(21160): eglMakeCurrent: 0xa17a2f60: ver 2 0 (tinfo 0xa34b2cc0)

```

```
[✓] Flutter (Channel beta, v0.6.0, on Mac OS X 10.13.5 17F77, locale en-CZ)
    • Flutter version 0.6.0 at /Users/vladimir.minkin/Documents/SDKs/Flutter
    • Framework revision 9299c02cf7 (2 weeks ago), 2018-08-16 00:35:12 +0200
    • Engine revision e3687f70c7
    • Dart version 2.1.0-dev.0.0.flutter-be6309690f

[!] Android toolchain - develop for Android devices (Android SDK 28.0.2)
    • Android SDK at /Users/vladimir.minkin/Library/Android/sdk
    • Android NDK at /Users/vladimir.minkin/Library/Android/sdk/ndk-bundle
    • Platform android-28, build-tools 28.0.2
    • ANDROID_HOME = /Users/vladimir.minkin/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1024-b01)
    ✗ Android license status unknown.

[✓] iOS toolchain - develop for iOS devices (Xcode 9.4.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 9.4.1, Build version 9F2000
    • ios-deploy 1.9.2
    • CocoaPods version 1.5.3

[✓] Android Studio (version 3.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1024-b01)

[!] VS Code (version 1.26.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension not installed; install from
      https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected devices (2 available)

```

**Comment by akeating** [comment]
Trying out Flutter for the first time. Plugin works fine on ios, but I can confirm I get this exact same exception on Android.

**Comment by knewter** [comment]
This error occurs because you do not have the app registered and the `google-services.json` file in place. At least, that's what I ran into tonight :)

In a tangentially related note, I cannot seem to catch the `PlatformException`s. I've seen other people reporting problems but no real solution. Tried a normal `try/catch` and a `try/on PlatformException catch/catch` and in both cases it just crashes the app and I don't catch it :-\
